smarts/core/scenario.py

```diff
@@ -263,24 +265,31 @@ def variations_for_all_scenario_roots(
             routes = Scenario.discover_routes(scenario_root) or [None]
             agent_missions = agent_missions or [None]
             social_agents = social_agents or [None]
+            traffic_histories = Scenario.discover_traffic_histories(scenario_root) or [
```

Comment:
I am considering getting rid of the `[{}]` in the `discover_traffic_histories() or [{}]`, i.e., we don't handle the edge case(`discover_traffic_histories` returns empty list `[]` ) here, maybe `traffic_histories = Scenario.discover_traffic_histories(scenario_root) or [None]` is better since it's aligned with the code above, and let the history provider to handle the case when history is `None`.